modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectDefinitionMVCActionCommand.java

```diff
@@ -117,44 +118,97 @@ private void _importObjectDefinition(ActionRequest actionRequest)
-			objectDefinitionJSONObject.toString());
+		JSONArray errorsJSONArray = _jsonFactory.createJSONArray();
 
-		objectDefinition.setActive(false);
+		for (Object object : jsonArray) {
+			JSONObject jsonObject = (JSONObject)object;
 
-		String externalReferenceCode = ParamUtil.getString(
-			actionRequest, "externalReferenceCode");
+			jsonObject.remove("accountEntryRestrictedObjectFieldId");
 
-		if (Validator.isNotNull(externalReferenceCode)) {
-			objectDefinition.setExternalReferenceCode(externalReferenceCode);
-		}
+			ObjectDefinition objectDefinition = ObjectDefinition.toDTO(
+				jsonObject.toString());
+
+			objectDefinition.setActive(false);
+
+			String externalReferenceCode = ParamUtil.getString(
+				actionRequest, "externalReferenceCode");
+
+			if (Validator.isNotNull(externalReferenceCode)) {
+				objectDefinition.setExternalReferenceCode(
+					externalReferenceCode);
+			}
+
+			String name = ParamUtil.getString(actionRequest, "name");
+
+			if (Validator.isNotNull(name)) {
+				objectDefinition.setName(name);
+			}
+
+			objectDefinition.setObjectFolderExternalReferenceCode(
+				ParamUtil.getString(
+					actionRequest, "objectFolderExternalReferenceCode"));
+
+			try {
+				ObjectDefinition putObjectDefinition =
+					objectDefinitionResource.
+						putObjectDefinitionByExternalReferenceCode(
+							objectDefinition.getExternalReferenceCode(),
+							objectDefinition);
+
+				putObjectDefinition.setPortlet(objectDefinition.getPortlet());
 
-		objectDefinition.setName(ParamUtil.getString(actionRequest, "name"));
-		objectDefinition.setObjectFolderExternalReferenceCode(
-			ParamUtil.getString(
-				actionRequest, "objectFolderExternalReferenceCode"));
+				if (!FeatureFlagManagerUtil.isEnabled("LPS-135430")) {
+					putObjectDefinition.setStorageType(StringPool.BLANK);
+				}
 
-		ObjectDefinition putObjectDefinition =
-			objectDefinitionResource.putObjectDefinitionByExternalReferenceCode(
-				objectDefinition.getExternalReferenceCode(), objectDefinition);
+				objectDefinitionResource.putObjectDefinition(
+					putObjectDefinition.getId(), putObjectDefinition);
+			}
+			catch (Exception exception) {
+				if (listImport) {
+					errorsJSONArray.put(
+						JSONUtil.put(
+							"error",
+							_handleImportException(actionRequest, exception)
+						).put(
+							"objectDefinitionName", objectDefinition.getName()
+						));
+				}
+				else {
+					errorsJSONArray.put(
+						_handleImportException(actionRequest, exception));
+				}
+			}
+		}
 
-		putObjectDefinition.setPortlet(objectDefinition.getPortlet());
+		if (errorsJSONArray.length() == 0) {
+			return null;
+		}
 
-		if (!FeatureFlagManagerUtil.isEnabled("LPS-135430")) {
-			putObjectDefinition.setStorageType(StringPool.BLANK);
+		if (!listImport) {
+			return errorsJSONArray.getJSONObject(0);
 		}
 
-		objectDefinitionResource.putObjectDefinition(
-			putObjectDefinition.getId(), putObjectDefinition);
+		return JSONUtil.put(
+			"message", errorsJSONArray
```

Comment:
The `message` is the keyword the front end uses for rendering its modals. Changing this to anything else could be costly. 
@danvb112 can probably explain this better if needed

Reply:
@G-Sa the problem is that `errorsJSONArray` is not a message, so it looks confusing

Reply:
maybe change to `errorsMessageJSONArray` helps?

Reply:
Hey guys, `message` is one of the defaults data property of an [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/message) instance. So to maintain the JS code convention, it's better to keep it. But do you think we can convert these JSONArray errors to a string or something similar, if so, would it be more acceptable to pass a string as a `message` @carolmariaabb?

Reply:
Or we can pass this `errorsJSONArray` as a `detail` instead of a `message`, the frontend would accept this, but I will have to modify somethings in my logic. What do you think?

Reply:
We ended up renaming the variable so it makes more sense, while keeping the behavior